smartlogic/client.go

```diff
@@ -102,7 +102,9 @@ func (c *Client) GetConcept(uuid string) ([]byte, error) {
 	// we additionally validate the response in order to check whether the response is for existing concept.
 	ok, err := c.isExistingConcept(body)
 	if err != nil {
-		return nil, fmt.Errorf("invalid concept representation returned for uuid %v", uuid)
+		errorMsg := fmt.Errorf("invalid concept representation returned for uuid %v", uuid)
```

Comment:
[Errorf](https://golang.org/pkg/fmt/#Errorf) returns an object that satisfies `error` interface. But looking at the code I suspect that we wanted to create a `string` rather than an `error`. We can use [Sprintf](https://golang.org/pkg/fmt/#Sprintf) for that.

Reply:
Errorf is used to create a new error object which is then returned as specified in the return type of the enclosing method.

Reply:
> Errorf is used to create a new error object which is then returned as specified in the return type of the enclosing method.

True. My suggestion was:
```suggestion
		errorMsg := fmt.Sprintf("invalid concept representation returned for uuid %s", uuid)
		entry.WithError(err).Error(errorMsg)
		return nil, errors.New(errorMsg)
```

In practice the code in the PR and the suggested one will have the same result. Both will create error log entry with `invalid concept representation returned for uuid %s` message.

The difference is that in the PR code we expect the `Error` method to convert the `errorMsg` from object to string, which could have undesired effects.